_episodes/l1-04-managing_contributions.md

```diff
@@ -0,0 +1,154 @@
+
+> ## Forking a repository
+> Let's have a go at forking the *book_of_recipes* repository on the Imperial College London GitHub organisation.
+> 1. First, navigate to the repository at [https://github.com/ImperialCollegeLondon/book_of_recipes](https://github.com/ImperialCollegeLondon/book_of_recipes) and in the top-right corner click **Fork**.
+> ![Fork button]({{ site.baseurl }}/fig/fork_button.png "Pull request title and description fields and create pull request button"){:class="img-responsive"}
+> 2. Select an owner for the forked repository (if you belong to any GitHub organisations, they will appear here) and give it a suitable name.
+> ![Create a fork with repository name emphasised]({{ site.baseurl }}/fig/fork_choose_owner_name.png "Create a fork with repository name emphasised"){:class="img-responsive"}
+> 3. Adding a description for your fork is optional. There is also a checkbox asking if you want to copy only the default branch of the repository (in this instance this is called `master`) or whether you want to copy all the branches. In most cases you will only want to copy the default branch. This option is selected by default. Finally, click **Create fork**.
+> ![Create a fork with description and create button]({{ site.baseurl }}/fig/fork_description_create.png "Create a fork with description and create button"){:class="img-responsive"}
+{: .challenge}
+
+After forking the repository, the second step is to make our fix/changes. First we will need to clone **our fork** so that we have the files in that repository locally on our computer (`clone` command was covered in the [introductory course](https://imperialcollegelondon.github.io/introductory_grad_school_git_course/l2-02-remote_repositories/index.html)). From here we can go ahead and create a new fix/feature branch and make our changes. When we are happy with the changes we have made, we can `commit` and `push` our upstream, forked repository.
+
+The third and final step in the workflow is to create a pull request. This is done in the same way as in the shared repository model above (navigate to your forked repository, click on the "Contribute" drop-down menu, then click the "Open pull request" button), only this time instead of the base branch being one in your repository, it is a branch in the upstream repository that you forked.
+
+![Drop-down menus for choosing the base fork and branch]({{ site.baseurl }}/fig/fork_pr_choose_branch.png "Drop-down menus for choosing the base fork and branch"){:class="img-responsive"}
+
+Another difference with pull requests from forked repositories is that you can allow anyone with push access to the upstream repository to make changes to your pull request. This is done by selecting **Allow edits from maintainers**.
+
+![Allow maintainers to make edits checkbox]({{ site.baseurl }}/fig/maintainer_edits.png "Allow maintainers to make edits checkbox"){:class="img-responsive"}
+
```

Comment:
Again, Atlassian has a nice [Forking Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow) example if we want to point them in that direction.